### PR TITLE
Add env option to swicth environment.

### PR DESCRIPTION
**Pull request by fossamagna**
I changed to contains multiple environments(scriptId) in `.clasp.json`.
Environment is specify by `env` option. If `env` option isn't specify, used as environment name is `default`.

refs #12

### CODE REVIEW

**Comment by fossamagna** [comment]
I signed it!

File: index.ts [hunk 1]
```diff
@@ -97,6 +103,50 @@ interface LoginOptions {
   localhost: boolean;
 }
 
+interface CreateOptions {
+  env: string;
```
Comment (grant) [inline index.ts]: We can't just add env options to every command. They need to be in `.clasp.json`.
https://github.com/google/clasp/issues/12#issuecomment-370672417

File: index.ts [hunk 2]
```diff
@@ -81,6 +83,10 @@ interface ClaspSettings {
   expiry_date: string;
 }
 // Project settings file (Saved in .clasp.json)
+interface ClaspProjectSettings {
```
Comment (grant) [inline index.ts]: `ClaspProjectSettings` is not needed. How is it different than `ProjectSettings`?

Comment (fossamagna) [reply index.ts]: `ClaspProjectSettings` is object to enclose `ProjectSettings` and has multiple `ProjectSettings`.
In https://github.com/google/clasp/issues/12#issuecomment-370672417, `ClaspProjectSettings` equivalent to `ProjectSettings` and `ProjectSettings` equivalent to `Env`.

File: index.ts (hunk 1 above)
Comment (fossamagna) [reply index.ts]: I think that `env` option is need to specify envId in `.clasp.json`. If no `env` option, How do it specify `envId` from CLI ?

Comment (grant) [reply index.ts]: I don't support a --env flag in each CLI command. Follow https://github.com/google/clasp/issues/12#issuecomment-370672417 carefully. It's very complex and I don't know if I covered all the cases. We should probably discuss more before implementing.

File: index.ts (hunk 2 above)
Comment (grant) [reply index.ts]: Can you use the exact terms as in the comment? I wrote:

```ts
// Project settings file (Saved in .clasp.json)
interface ProjectSettings {
  scriptId: string; // default env
  rootDir: string;  // default env
  envs: Map<EnvId, Env> // custom scriptIds (NEW)
}
```

`ClaspProjectSettings` is redundant. Either say `ClaspSettings` or `ProjectSettings`.

Comment (fossamagna) [reply index.ts]: I see. I will use extact terms.

File: index.ts (hunk 1 above)
Comment (fossamagna) [reply index.ts]: I understood your idea.
I have a consideration for use-case of multiple google accounts and multiple `scriptId`s at same time. Please see https://github.com/google/clasp/issues/12#issuecomment-371168118

**Comment by grant** [comment]
I don't think this PR is getting closer to adding environments to clasp projects. Do you mind if we close this PR and discuss more in #12?

**Pull request closed without merging**